lib/tbd/ua.rb

```diff
@@ -183,151 +187,163 @@ def uprate(model = nil, s = {}, argh = {})
              g[:op].downcase == "all floor constructions"
 
       if g[:op].empty?
-        log(ERR, "Construction to uprate? (#{mth})")
+        log(ERR, "Construction (#{type}) to uprate? (#{mth})")
       elsif all
-        model.getSurfaces.each do |sss|
-          next unless sss.surfaceType.downcase.include?(label.to_s)
-          next unless sss.outsideBoundaryCondition.downcase == "outdoors"
-          next     if sss.construction.empty?
-          next     if sss.construction.get.to_LayeredConstruction.empty?
-
-          c         = sss.construction.get.to_LayeredConstruction.get
-          i         = c.nameString
-
-          # Reliable unless referenced by other surface types e.g. floor vs wall.
-          if c.getNetArea > area
-            area = c.getNetArea
+        s.each do |nom, surface|
+          next unless surface.key?(:deratable   )
+          next unless surface.key?(:type        )
+          next unless surface.key?(:construction)
+          next unless surface.key?(:filmRSI     )
+          next unless surface.key?(:index       )
+          next unless surface.key?(:ltype       )
+          next unless surface.key?(:r           )
+          next unless surface[:deratable   ]
```

Comment:
Fix: simply rely on the TBD `:deratable` tag for insulated attic floors (to uprate).